src/widget-core/meta/InputValidity.ts

```diff
@@ -0,0 +1,26 @@
+import Base from './Base';
+
+export class InputValidity extends Base {
+	get(key: string | number, value: string) {
+		const node = this.getNode(key) as HTMLFormElement | undefined;
+
+		if (!node) {
+			return { valid: undefined, message: '' };
+		}
+
+		if (value !== node.value) {
+			// if the vdom is out of sync with the real dom our
+			// validation check will be one render behind.
+			// Call invalidate on the next loop.
+			setTimeout(() => this.invalidate());
+			return { valid: undefined, message: '' };
```

Comment:
could this not cause a flickering?

Reply:
I don't think you need to return her at all to be honest Nick, you weren't doing so in the POC.

Reply:
Oops, that should actually be removed.